rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -195,6 +195,15 @@ http {
         ''               $host;
     }
 
+    {{ if $cfg.ComputeFullForwardedFor }}
+    # We can't use $proxy_add_x_forwarded_for because the realip module
+    # replaces the remote_addr to soon
+    map $http_x_forwarded_for $full_x_forwarded_for {
+        default          "$http_x_forwarded_for, $realip_remote_addr";
+        ''               "$realip_remote_addr";
```

Comment:
realip_remote_addr does not contains the client IP address. Please use $the_real_ip

Reply:
That's on purpose. Here I want to append the ip of the actual tcp client connected to the Ingress.

The goal of this PR is to have the `X-Forwarded-For` header arriving on upstreams has if it had been through any classic L7 load-balancer, and the last part of it should be the last proxy the request went through.

Maybe my explanations are not clear enough, let me add an example.
```
Client (1.1.1.1) => L7 Load-balancer A (2.2.2.2) => L7 Ingress B (3.3.3.3) => Pods
```
The **Load-balancer A** appends its IP to the `X-Forwarded-For` header of the request, or create the header if it doesn't exist (which should be the case).

If `3.3.3.3` was another L7 load-balancer/reverse proxy, it would also append `2.2.2.2` to the value of `X-Forwarded-For` and the client would receive `X-Forwarded-For: 1.1.1.1, 2.2.2.2`.

But with the current Ingress, the Pod receives the request with the header `X-Forwarded-For: 1.1.1.1` if `proxy-real-ip-cidr: 2.2.2.2`. We loose the information that the request has been through `2.2.2.2`, expect if we look in `X-Original-X-Forwarded-For` but this is not standard.

With the PR, no matter the value of `proxy-real-ip-cidr`, the Pod would receive the request with the header `X-Forwarded-For: 1.1.1.1, 2.2.2.2`. The implementation of the pod could check the header, compare `2.2.2.2` to a list of trusted proxy and because it matches, set the real-ip to `1.1.1.1`, or rely of `X-Real-IP` directly.

Reply:
As you know, I would vote for having this the default behavior and `X-Original-Forwarded-For` removed :) 
But I understand it could break some older Ingress Nginx setup :/